app/models/extract_fetcher.rb

```diff
@@ -65,14 +50,36 @@ def fetch_subjects?
   end
 
   def user_extracts
-    @user_extracts ||= Extract.where(filter.except(:subject_id)).order(classification_at: :desc)
+    corrected_filter = filter.except(:subject_id)
+    if fetch_minimal?
+      Extract.where(corrected_filter.merge(id: @extract_ids))
+    else
+      Extract.where(corrected_filter)
+    end
   end
 
   def subject_extracts
-    @subject_extracts ||= Extract.where(filter.except(:user_id)).order(classification_at: :desc)
+    extract_subject_ids = Extract.find(@extract_ids).pluck(:subject_id)
+    exact_subject_ids = extract_subject_ids.append(filter[:subject_id]).uniq
+    augmented_subject_ids = augment_subject_ids(exact_subject_ids)
+
+    corrected_filter = filter.except(:user_id)
+
+    if fetch_minimal?
+      # is an extract exactly in the list of extracts with the specified subject
+      # or is an extract for one of those prior subjects but not the specified subjects
+      Extract.where(corrected_filter.merge(id: @extract_ids))
```

Comment:
I find these very hard to interpret - could they be broken up to named scopes or methods to help reason about this scope lookup?

As you are switching the behaviour of this method based on the `fetch_minimal?` state, ideally these conditional blocks call named methods to help me reason about the different expected behaviour.

Reply:
i'm not sure it improves the clarity much but i made this change